### ISSUE TITLE
Feature request: strict/fail-fast mode 

### ISSUE THREAD

**Issue by savente93** [cargo-generate/cargo-generate#1339]
This is more of a feature request for the template authors than the users. I started making my own templates a few days ago, and had quite a few head scratches why my template files weren't being expanded at all. Turns out I made a syntax error, which I would have caught sooner if I had used the `--verbose` flag sooner which was on me. 

However, I do feel like this highlights two things I think a lot of folks like me could benefit from:
1. I think that template syntax error that prevent template expansion completely (at least within one a file) should raise a warning even in non-verbose mode. Even as a user trying to use a broken template I think being told about this could save a lot of head scratches. 
2. I'd appreciate a fail-fast or struck mode that halts execution on errors. Part of this is there in `--silent` which fails if not all variables are supplied, but there isn't anything similar for the liquid side as far as I can tell. Whether we should clean up after a failed run (i.e. remove output that was generated previously in a failed run) I don't actually care that much, you could argue for either. Happy to go with whatever maintainers think is best here. 

If this is something the project is interested in I'd be open to making a PR for this myself, but wanted to discuss it first.

**Comment by sassman** [comment]
I'm not sure if you have already seen the `--test` argument. It's meant to simplify the template authors life already a bit.

However, I think some checks, for example syntax errors within the liquid template tags, could also be helpful to run on the `--test` run.

IMO there is quite some room for improvemnt.

**Comment by savente93** [comment]
yeah, I did see `--test` but since I wasn't getting any output I figured that that wouldn't be too helpful, since it only mentioned running `cargo test` and I didn't think that would help in my specific case. But I do think this could be a good place for it. 

so far my list is:
- [ ] raising a warning on syntax errors in the template if they exist
- [ ] fail the process in that case if we are running in `--test` mode

Anything else that should be included?
